synergy-design-system/synergy-design-system
# feat: ✨ Implement dark mode for components and documentation

# Pull Request

## 📖 Description

This PR introduces the first version of our components dark mode. It applies changes to `@synergy-design-system/tokens`, as well as our documentation to accomplish this.

It also comes with support for [Chromatic modes](https://www.chromatic.com/docs/themes/), that will make sure each story will be captured in light AND dark mode automatically.

### 🎫 Issues

Closes #147.

## 👩‍💻 Reviewer Notes

Just have a look at the theme toggle. Storybooks overall appearance should not change at all. You are now able to toggle between `🌞 Synergy (light)` (the default) and `🌙 Synergy (dark)`. I have opted to prefix the theme name with `Synergy` instead of going just with light and dark, as we **may** add other themes later on.

I have also removed some clutter from `docs.css` that was unused. It now uses only the bare minimum to style the content. Also, the backgrounds addon was altered to make sure it is still usable for checking backgrounds and so on.

Documentation for `@synergy-design-system/tokens` was also updated to reflect those changes. Also, we now ship demos for angular, react and vue that show how themes can be toggled.

## 📑 Test Plan

- Check out the branch and issue `pnpm install -r`
- Run a build to make sure everything is setup correctly (`pnpm build` in the root)
- Start storybook and navigate to a docs page or story.
- Toggle the selection of `🌞 Synergy (light)` to `🌙 Synergy (dark)`. You should now see a dark  mode version of the documentation snippet (in case of docs) or the full preview (on a story).
- Toggle back to light mode to see the change.

## ✅ DoD
